eth/tracers/api.go

```diff
@@ -269,10 +269,11 @@ func (api *API) traceChain(start, end *types.Block, config *TraceConfig, closed
 				var (
 					signer   = types.MakeSigner(api.backend.ChainConfig(), task.block.Number(), task.block.Time())
 					blockCtx = core.NewEVMBlockContext(task.block.Header(), api.chainContext(ctx), nil)
+					rules    = api.backend.ChainConfig().Rules(task.block.Number(), task.block.Difficulty().BitLen() == 0, task.block.Time())
```

Comment:
I think we should do something like this. Everywhere else in the codebase we use random != nil to determine the post-merge blocks

```suggestion
					rules    = api.backend.ChainConfig().Rules(blockCtx.BlockNumber, blockCtx.Random != nil, blockCtx.Time)
```

Reply:
I think we can change to `Random != nil` where it makes sense, but there are some cases where it is not easy to e.g. `traceBlockParallel` where the block context is not safe for concurrent access and gets redefined each iteration. Yeah we can usually construct a block context that is only used for this purpose, but under the hood it also is just checking if Difficultly is zero under the hood.

Maybe it makes sense to create another function `RulesFromHeader` or a helper `IsPostMerge`. But I think creating block contexts just to check if it is post merge is unnecessary.